zstacklib/zstacklib/network/firewall/iptables.py

```
@@ -0,0 +1,536 @@
+        table.delete_child_by_name(chain_name)
+
+
+def from_iptables_save() -> IPTables:
+    """Create IPTables instance from current iptables-save output."""
+    return IPTables.from_iptables_save()
+
+
+def insert_single_rule_to_filter_table(rule: str) -> None:
+    """Insert a single rule to filter table if not exists."""
+    if not rule.startswith('-A '):
+        raise IPTablesError("rule must start with '-A '")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)